python/cudf/cudf/core/index.py

```diff
@@ -466,8 +487,36 @@ def __init__(self, start, stop=None, name=None):
             start, stop = 0, start
         self._start = int(start)
         self._stop = int(stop)
-        self.name = name
         self._cached_values = None
+        self._index = None
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    def as_column(self):
+        if len(self) > 0:
+            vals = cudautils.arange(self._start, self._stop, dtype=self.dtype)
+        else:
+            vals = rmm.device_array(0, dtype=self.dtype)
```

Comment:
Use empty buffer constructor instead of `device_array`

Reply:
Done. Used `column_empty` instead